pennylane/devices/default_qubit.py

```diff
@@ -59,6 +59,10 @@ class DefaultQubit(Device):
             using a pool of at most ``max_workers`` processes. If ``max_workers`` is ``None``,
             only the current process executes tapes. If you experience any
             issue, say using JAX, TensorFlow, Torch, try setting ``max_workers`` to ``None``.
+        prng_key (Optional[jax.random.PRNGKey]): An optional ``jax.random.PRNGKey``. Defaults to None.
```

Comment:
We also have `seed`. What is the motivation for an extra argument?

Reply:
1. default.qubit.jax also has `prng_key`, so it's consistent there
2. Lillian first attempted to use the value for `seed` as the PRNG key input if it's an int, but it was too messy

Concern for messiness:

`seed` can be an int, but it can also be a whole bunch of other types that ultimately become a numpy RNG. this was straightforward before, because regardless of what type the user passes in, we initialized an rng and [saved that](https://github.com/PennyLaneAI/pennylane/blob/cbc8d19e6a36f0d4d38b44eabd4a8b98db677327/pennylane/devices/default_qubit.py#L151) as the device property. It's then used in a variety of places as needed. It would degrade code quality to write each of these things twice without a better abstraction, and writing that feels overkill when adding a kwarg (that was already there before) is pretty straightforward.

`prng_key` was only used in one place - generating samples. It also behaved in a strange way, which is that if you have a jax device with a PRNG key saved to it, and ran the same circuit twice, it would re-create the same samples (because it would use the same initial seed per-execution). `seed` doesn't work that way in the new device, nor should it, because this was a jax-PRNG-key-specific behaviour.

Lastly, because of interface unwrapping and other issues, it can be hard to know _when_ the user wants regular numpy RNG, or jax PRNG. This way, it's much clearer. When generating samples, do I have a PRNG key saved?
- yes? use it
- no? use numpy rng

I think that pretty much covers it 😄